vendor/0install-solver/src/solver/dune

```diff
@@ -0,0 +1,2 @@
+(library
+ (name zeroinstall_solver))
```

Comment:
Do we want to call it `dune_zeroinstall_solver` like we do for other vendored libs?

Reply:
We don't seem to consistently apply that convention to all our vendored libs. What's the reason for renaming them with the `dune_` prefix?

Reply:
it's explained here https://github.com/ocaml/dune/pull/7179#issuecomment-1497848152

It's done on a per-need basis. If it's not causing conflicts, then let it be as is.

Reply:
Ah cool. Would it be helpful to add that comment or words to the same effect to a readme file in dune's vendor directory so it's more clear what to do next time someone needs to vendor something?